apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/interceptor/TokenProvider.kt

```diff
@@ -0,0 +1,6 @@
+package com.apollographql.apollo.interceptor
+
+interface TokenProvider {
+  suspend fun currentToken(): String
+  suspend fun renewToken(previousToken: String): String
```

Comment:
I wonder how much we should follow OAuth standards here? I was thinking about refreshToken for example. That is usually used for getting a new token in such scenario.

Reply:
Renamed. Also, that aligns perfectly with `currentToken` 👌